docs/docs/CPG/specs/inference.md

```diff
@@ -8,20 +8,31 @@ description: >
     Inference of new nodes
 ---
 
+# Inference System
+
 One of the goals of this library is to deal with incomplete code. In
 this case, the library provides various options to create new nodes
 and include them in the resulting graph. The user of the library can
 configure which of the inference options should be enabled. This
 document provides an overview of the different options and their
 expected behavior. The rules for the inferring new nodes are implemented
-in the class `de.fraunhofer.aisec.cpg.passes.inference.Inference` and
+in the class [`de.fraunhofer.aisec.cpg.passes.inference.Inference`](https://fraunhofer-aisec.github.io/cpg/dokka/main/older/main/cpg-core/de.fraunhofer.aisec.cpg.passes.inference/-inference/index.html) and
 are typically used by various passes.
 
-# Inference of namespace declarations
+## Inference of namespace declarations
+
+If we encounter a scope, e.g, in a call to a function such as 
+`java.lang.Object.toString()`, and we do not have a corresponding `NameScope` 
+for the qualified name `java.lang`, we try to infer one. We recursively infer
+a namespace, e.g., `java` as well as `java.lang` until the scope can be resolved.
+There is one special check, in case the name refers to a type. In this case we 
+infer a record declaration instead. This is usually the case when a type is nested
+in another type, e.g. `MyClass::MyIterator::next`. If we encounter usage of `MyClass::MyIterator`
+as a type somewhere, we infer a record instead of a namespace.
 
-# Inference of record declarations
+## Inference of record declarations
```

Comment:
This description is missing. Probably it's simply the last part of the previous paragraph? Merge the two headings?